File: backend/services/tts_service.py
import os
import requests
from pathlib import Path
from gtts import gTTS


async def generate_audio(job_id: str, script: dict) -> Path:
    """Generate audio from script using gTTS (Google Text-to-Speech)"""
    
    text = script.get("full_text", "")
    
    try:
        # Using gTTS - free and simple
        tts = gTTS(text=text, lang='en', slow=False)
        
        audio_path = Path("temp") / f"{job_id}_audio.mp3"
        tts.save(str(audio_path))
        
        return audio_path
        
    except Exception as e:
        raise Exception(f"TTS generation failed: {str(e)}")

File: backend-node/services/ttsService.js
import gtts from 'gtts';
import fs from 'fs';
import path from 'path';

export async function generateAudio(jobId, script) {
  const text = script.full_text || '';
  
  return new Promise((resolve, reject) => {
    const audioPath = path.join('temp', `${jobId}_audio.mp3`);
    
    // Use Hindi language for better Hinglish pronunciation
    const speech = new gtts(text, 'hi');
    
    speech.save(audioPath, (err) => {
      if (err) {
        reject(new Error(`TTS generation failed: ${err.message}`));
      } else {
        resolve(audioPath);
      }
    });
  });
}
